nemo/nemo/core/neural_modules.py

```diff
@@ -89,6 +82,22 @@ def __init__(
         #        "arguments:".format(self.__class__.__name__))
         #    self._logger.warning("{}".format(kwargs.keys()))
 
+    @abstractmethod
+    def input_port_definitions(self) -> Optional[Dict[str, NeuralType]]:
```

Comment:
I remember talking abut this, but can you remind me why this is a method instead of a property?

Reply:
Delete  "input_port_definitions" and "output_port_definitions". You can keep using "input_ports" and "output_ports" properties instead. 
This will require much fewer changes throughout the codebase and will make the transition easier.

Reply:
Hey, so this is "just a matter of taste."
https://stackoverflow.com/questions/20506612/python-property-vs-method-when-no-access-to-attribute-is-needed

IMO in several cases we are performing some actual calculations there (e.g. generating list of ports etc.), so I decided that those getters should be methos.

But if majority of you want it to be changed into property, I have no problem with
replacing @abstractmethod with @abstractproperty ;)

Reply:
it is not a matter of taste. If it is a property one could use "myclass.input_ports" not "myclass.input_ports()". We aren't going to pass any args there, so it should be property then.